cluster-api-gcp/deploy/deploy_helper.go

```diff
@@ -53,7 +54,7 @@ func (d *deployer) createCluster(c *clusterv1.Cluster, machines []*clusterv1.Mac
 	}
 
 	if master.GetName() == "" {
-		master.Name = master.GetGenerateName() + c.GetName()
+		master.Name = names.SimpleNameGenerator.GenerateName(master.GetGenerateName()) + "-" + c.GetName()
```

Comment:
IIUC, it seems we get the name generation for nodes as we go through the API server. Here, however, we're going through the machine deployer (which extends the machine actuator) and we're using the cloud provider code directly.

Can you please add a comment explaining why the name generation for the master is needed here?